ci.sh

```diff
@@ -100,6 +100,10 @@ else
     mkdir empty || true
     cd empty
 
+    # get mypy tests a nice cache
+    # (this ugly env var is necessary until we have a py.typed file)
+    MYPYPATH=".." mypy --config-file= --cache-dir=./.mypy_cache -c "import trio" >/dev/null 2>/dev/null || true
+
```

Comment:
Since the long-term plan is to have one anyway, we could just temporarily create one here in `ci.sh` (and remove it at the end if it was created).

I had that previously but switched to creating it in the tests themselves so they could be run directly through pytest w/o having to manually create a `py.typed`, but not much reason not to do both.